### PR TITLE
ShopExtension - adding Category fields

### LINKED ISSUE #653: Shop-extension Category type is missing important fields
When requesting a specific Category on GQL - `urlPath` is not present:

```graphql
query Category {
  category(id:"19") {
    id
    name
  }
}
```

It would be nice to have:
- `urlPath`
- `image`
- `attributes`

### PR DESCRIPTION
**Pull request by Borales**
### Please check if the PR fulfills these requirements ###
Adding basic fields to Category type in Shop Extension schema (and resolvers for magento2-api)

- [x] The commit message follows our guidelines
- [x] Changelog have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**
Improvement

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**
No

**Other information:**
Fixes #653

### CODE REVIEW

File: packages/falcon-front-kit/src/SEO/SEO.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet-async';
+import { SEO as SEOType } from '@deity/falcon-shop-extension';
+
+export type SeoProps = {
+  meta: Partial<SEOType>;
+  title: string;
+};
+
+export const SEO: React.FC<SeoProps> = ({ meta: { title: metaTitle, description, keywords }, title }) => (
```
Comment (nimobeeren) [inline packages/falcon-front-kit/src/SEO/SEO.tsx]: Why do we need both `title` and `meta.title`?

**Comment by Borales** [comment]
@nimobeeren I pushed the fix (weird, it wasn't related to my changes). Btw, the query should have another nested level:

```graphql
productList {
  items {
    id
    categories {
      name
    }
  }
}
```

File: packages/falcon-front-kit/src/SEO/SEO.tsx (hunk 1 above)
Comment (Borales) [reply packages/falcon-front-kit/src/SEO/SEO.tsx]: @nimobeeren `title` is something that a developer needs to provide manually (and it's a required prop). `meta` may contain a custom `meta.title` (set for SEO-purposes), but if `meta` doesn't provide one - `title` will be used for this purpose, so we'll be having at least 1 required title.

File: packages/falcon-magento2-api/src/index.ts [hunk 2]
```diff
@@ -320,8 +326,9 @@ module.exports = class Magento2Api extends Magento2ApiBase {
    * @returns {Promise<Product[]>}  response with list of products
    */
   async productList(obj, { input }) {
-    const { withAttributeFilters = [] } = input;
-    const searchCriteria = this.createSearchCriteria(input);
+    // It's not being used anymore.
+    const withAttributeFilters = [];
```
Comment (mateuszwozniak) [inline packages/falcon-magento2-api/src/index.ts]: So just remove this variable and put `withAttributeFilters: []` in the actual call (if Magento requires it).

Comment (Borales) [reply packages/falcon-magento2-api/src/index.ts]: removed

File: packages/falcon-shop-extension/src/types.ts [hunk 3]
```diff
@@ -334,7 +335,7 @@ export type ConfigurableProductOptionValue = {
   valueIndex: string;
 };
 
-export type ProductSeo = {
+export type SEO = {
```
Comment (tutok) [inline packages/falcon-shop-extension/src/types.ts]: should not we add also fields like `image` or `urlPath`? these are also necessary from SEO perspective

File: examples/shop-with-blog/client/src/pages/shop/Category/Category.js [hunk 4]
```diff
@@ -38,9 +37,7 @@ const CategoryPage = ({ match: { params } }) => (
 
           return (
             <CategoryLayout variant={!filtersData.length && 'noFilters'}>
-              <Helmet>
-                <title>{name}</title>
-              </Helmet>
+              <SEO meta={category.seo} title={name} />
```
Comment (tutok) [inline examples/shop-with-blog/client/src/pages/shop/Category/Category.js]: > should not we add also fields like image or urlPath? these are also necessary from SEO perspective

(referring to this comment)

`category` (I think `Product` or `BlogPost` also) contains more SEO information like mentioned above `image`. So maybe we could introduce interface (`ISeoEntity`) which could be later implemented via `Category`, `Product`, `BlogPost`... then above line could look like:
```suggestion
              <SEO {...category} />
```

I think Interface is better than duplicate all fields into `entity.seo` node

File: packages/falcon-shop-extension/src/types.ts (hunk 3 above)
Comment (Borales) [reply packages/falcon-shop-extension/src/types.ts]: @tutok this task was mainly focused on adding missing fields to Category, and in order to re-use some existing types - I renamed `ProductSeo` to `SEO` (it wouldn't make sense to use "Product" prefix for Category).

There's even much more stuff related to SEO for products, like price, stock, reviews, brand etc - but I think this deserves to have a separate task to fully cover SEO needs

File: examples/shop-with-blog/client/src/pages/shop/Category/Category.js (hunk 4 above)
Comment (Borales) [reply examples/shop-with-blog/client/src/pages/shop/Category/Category.js]: @tutok the proposed idea is great, but I'm afraid merging fields into the main type could lead to unwanted behavior like having a need to use a custom title for `<title>` tag (for example "Buy these cool BrandName X100 headphones"), while keeping `<h1>` more simple for visitors (for example "BrandName X100 headphones") - this way `title` is not a duplicate and serves different purposes. Same goes to `description` - `description` of the Category may contain a fully-featured HTML content, while `seo.description` is limited (by length) and must be a plaintext.

Besides that, keeping SEO-related stuff under `seo` key lets other developers using this data from another source (like external SEO-tool) while keeping the main Product/Category data to be fetched from the shop-backend.

**Pull request merged**